src/runtime/StringObject.cpp

```diff
@@ -48,6 +48,11 @@ void* StringObject::operator new(size_t size)
     return GC_MALLOC_EXPLICITLY_TYPED(size, descr);
 }
 
+bool StringObject::hasProperty(ExecutionState& state, const ObjectPropertyName& P) ESCARGOT_OBJECT_SUBCLASS_MUST_REDEFINE
+{
+    auto r = getOwnProperty(state, P);
```

Comment:
Why did you add this `hasProperty` method for StringObject and others? According to the ES2015 spec, it seems that there are only 2 defined `hasProperty` methods which are for `Object` and `Proxy` only

Reply:
It's all about overriding and fast cases. Since each special object implements it's own GetOwnPropety routine while overriding the Object::getOwnProperty, IMO we should do the same for hasProperty as well.

Reply:
It's all about overriding and fast cases. Since each special object implements it's own GetOwnPropety routine while overriding the Object::getOwnProperty, IMO we should do the same for hasProperty as well.